### PR TITLE
DOC: Review and expand docstrings for np.gradient() 

### LINKED ISSUE #17867: DOC: Review and expand docstrings for np.gradient()
### Reproducing code example:

```python
import numpy as np
x = np.array([0, 1, 2, 3, 4])
dx = np.gradient(x)
y = x**2
np.gradient(y, dx, edge_order=2)
```

### Error message:

```
array([nan, inf, inf, inf, nan])
```

shouldn't it be the following?

```
array([-0.,  2.,  4.,  6.,  8.])
```

### NumPy/Python version information:

```
1.19.4 3.8.3 (default, Jul  2 2020, 11:26:31) 
[Clang 10.0.0 ]
```

### PR DESCRIPTION
**Pull request by rajveer43**
Close #17867

### CODE REVIEW

File: numpy/lib/function_base.py [hunk 1]
```diff
@@ -1074,6 +1074,28 @@ def gradient(f, *varargs, axis=None, edge_order=1):
     array([[ 2.,  2., -1.],
            [ 2.,  2., -1.]])
 
+    >>> x = np.array([1 + 2j, 2 + 4j, 3 + 6j, 4 + 8j])
+    >>> np.gradient(complex_array.real)
```
Comment (rgommers) [inline numpy/lib/function_base.py:1078]: No output is shown, so this line isn't useful. Delete it?

File: numpy/lib/function_base.py [hunk 2]
```diff
@@ -1074,6 +1074,28 @@ def gradient(f, *varargs, axis=None, edge_order=1):
...
+    >>> np.gradient(complex_array.imag)
+    array([2., 2., 2., 2.])
+
+    >>> x = np.linspace(-1, 1, 5)
+    >>> y = np.linspace(-1, 1, 4)
+    >>> X, Y = np.meshgrid(x, y, indexing='ij')
+    >>> vector_field_x = -Y
+    >>> vector_field_y = X
+    >>> np.gradient(vector_field_x, x, y)
```
Comment (rgommers) [inline numpy/lib/function_base.py:1087]: same, not useful - please delete

**Review by rgommers — CHANGES_REQUESTED** [review]
Thanks @rajveer43. The test failure on Azure is real, can you please fix the example output @rajveer43?

**Comment by rkern** [comment]
I don't see how this relates to #17867. These seem to be random uses of `np.gradient()` without a pedagogical purpose.

**Comment by rkern** [comment]
Personally? None, to be honest. But the conversation in #17867 brought up these:

> Also some examples for np.gradient in this regard would be very helpful, specifically simple examples like f=x**2, f'x = 2*x, and how to obtain np.gradient to show exactly f'x = 2*x, or f = exp(x), f'x = exp(x).

**Comment by mdhaber** [comment]
Looks like this is no longer active, and it did not address the issue as recommended, so closing.

**Pull request closed without merging**